### PR TITLE
add option `--console.history` to arangosh

### PR DESCRIPTION
**Pull request by jsteemann**

### CODE REVIEW

File: arangosh/Shell/ConsoleFeature.cpp [hunk 1]
```diff
@@ -108,6 +109,10 @@ void ConsoleFeature::collectOptions(std::shared_ptr<ProgramOptions> options) {
   options->addOption("--console.audit-file",
                      "audit log file to save commands and results",
                      new StringParameter(&_auditFile));
+  
+  options->addOption("--console.history",
+                     "whether or not to load and persist command-line history",
+                     new BooleanParameter(&_useHistory));
```
Comment (Simran-B) [inline arangosh/Shell/ConsoleFeature.cpp]: I wonder if we should generally add the introducedIn information for new options from now on, even if it does not get backported to a GA version. @sleto-it What do you think?

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3336/

File: arangosh/Shell/ConsoleFeature.cpp (hunk 1 above)
Comment (sleto-it) [reply arangosh/Shell/ConsoleFeature.cpp]: I believe yes - absolutely yes. If this option is available only from 3.5, we should say it clearly imho, as usual. Thanks,

Comment (Simran-B) [reply arangosh/Shell/ConsoleFeature.cpp]: will be available from v3.4.5 and v3.5.0 and this information will also be in the docs. For old jenkins the options dump needs to be updated as per usual.

**Pull request merged**